hollow/src/main/java/com/netflix/hollow/core/write/HollowBlobWriter.java

```diff
@@ -280,13 +280,17 @@ private void writeHeaders(DataOutputStream os, ProducerOptionalBlobPartConfig.Op
 
         /// write main header
         HollowBlobHeader header = new HollowBlobHeader();
-        header.setHeaderTags(stateEngine.getHeaderTags());
+        header.getHeaderTags().putAll(stateEngine.getHeaderTags());
```

Comment:
subtle change here to make a deep copy

Reply:
Are we validating the size of header tags map in the UT?